pkg/applicationserver/io/pubsub/provider/mqtt/driver.go

```diff
@@ -58,6 +58,8 @@ func openDriverTopic(client mqtt.Client, topicName string, timeout time.Duration
 	return dt, nil
 }
 
+var errPublishFailed = errors.Define("publish_failed", "publish to MQTT topic failed")
```

Comment:
Is `Unknown` the best we can do here ? Would 'FailedPrecondition' be more appropriate ?

Reply:
`FailedPrecondition` would indicate that a retry is pointless, and we don't actually know.